### PR TITLE
mobile: using the API listener by default

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: medium
Testing: yes
Docs Changes: no
Release Notes: inline
Fixes https://github.com/envoyproxy/envoy-mobile/issues/2711

### CODE REVIEW

File: .github/workflows/core.yml [hunk 1]
```diff
@@ -29,6 +29,7 @@ jobs:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       run: |
         cd mobile && ./bazelw test \
+            --build_tests_only \
```
Comment (Augustyniak) [inline .github/workflows/core.yml:32]: just curious - was this needed or just an improvement?

File: mobile/test/common/jni/BUILD [hunk 2]
```diff
@@ -53,3 +53,35 @@ envoy_mobile_so_to_jni_lib(
     testonly = True,
     native_dep = "libenvoy_jni_with_test_extensions.so",
 )
+
+# Same as above but wtih listener extensions too.
```
Comment (Augustyniak) [inline mobile/test/common/jni/BUILD:57]:
```suggestion
# Same as above but with listener extensions too.
```

**Review by Augustyniak — APPROVED** [review]
+1 with one nit/typo and one question.

I went over the referenced GH issue to catch up on the decisions that were made with regard to the rollout of the change.

File: .github/workflows/core.yml (hunk 1 above)
Comment (alyssawilk) [reply .github/workflows/core.yml:32]: it avoids trying to build mobile/test/common/jni/BUILD on the build we rm -rf the listener file :-P

**Comment by alyssawilk** [comment]
ack - I'd rather wait a few days than land without testing - feel free to mark as waiting on in-house testing if needed!

**Comment by jpsim** [comment]
Lyft testing on iOS and Android is complete, everything looked good.

I'll let Rafal review this PR though.

**Review by Augustyniak — APPROVED** [review]
@alyssawilk The change looks good to me. Thank you for verifying @jpsim.

.

**Comment by Augustyniak** [comment]
aah, auto merge was enabled in here

**Comment by alyssawilk** [comment]
ugh sorry.  happy to roll back and resubmit next week if you'd prefer?

**Comment by jpsim** [comment]
> ugh sorry. happy to roll back and resubmit next week if you'd prefer?

I think I'm missing something here. As far as I know this hasn't caused any issues?

**Comment by Augustyniak** [comment]
>I think I'm missing something here. As far as I know this hasn't caused any issues?

It related to my previous (and edited) message where I suggested we wait with the merge until next week. I removed that message after I realized the PR was merged due to auto-merge enabled for the PR (and me approving it). @alyssawilk I think that we are fine here but thank you for offering to help. Apologies for the confusion. I will let you know if we run into any issues and need to revert.

**Pull request merged**